src/main.rs

```diff
@@ -41,26 +42,89 @@ struct Config {
 pub mod handlers;
 
 fn main() {
-    let args = Args::parse();
+    let args: Args = Args::parse();
     let raw_config = fs::read_to_string(args.config.as_str()).unwrap();
-    let mut config: Config = serde_json::from_str(raw_config.as_str()).unwrap();
+    let mut json_config: Value =
+        serde_json::from_str(&raw_config).expect("Failed to parse raw config");
+
+    // Ensuring caddy field is defined
+    if json_config.get("caddy").is_none() {
+        let default_caddy: Value = json!({
+            "url": "",
```

Comment:
Instead of using "", can we use None and make the struct fields Options when required.